### PR TITLE
Tag releases via PR

### PR DESCRIPTION
**Pull request by hakman**
After merging a PR that modifies version.txt, a tag will be created.
Similar to:
* https://github.com/kubernetes/cloud-provider-aws/blob/master/.github/workflows/tag.yml
* https://github.com/kubernetes/kops/blob/master/.github/workflows/tag-release.yml

/cc @vteratipally 
/assign @vteratipally

### CODE REVIEW

**Comment by MartinForReal** [comment]
I'm a little bit confused. Maybe It is easier if we push new tag to github?

**Comment by hakman** [comment]
> I'm a little bit confused. Maybe It is easier if we push new tag to github?

That is dependent on someone having admin permissions on the repo.
Doing it via PR is possible for any contributor to propose a new release. After merge, one more PR in the k8s.io repo and that should be it.
In any case, it's a proposal.

File: .github/workflows/tag-release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - version.txt
+
+jobs:
+  tag:
+    if: ${{ github.repository == 'kubernetes/cloud-provider-aws' }}
```
Comment (xiaods) [inline .github/workflows/tag-release.yml:12]: why cloud-provider-aws?

File: hack/tag-release.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  exit 1
+fi
+
+if [ "$(git tag -l "v${VERSION}")" ]; then
+  echo "Tag v${VERSION} already exists"
+  exit 1
+fi
+
+git tag -a -m "Release ${VERSION}" "v${VERSION}"
+git push origin "v${VERSION}"
```
Comment (xiaods) [inline hack/tag-release.sh:30]: suggest add v to version.txt

File: version.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+0.8.15
```
Comment (xiaods) [inline version.txt:1]: suggest add v to version.txt

File: .github/workflows/tag-release.yml (hunk 1 above)
Comment (hakman) [reply .github/workflows/tag-release.yml]: good point, fixed. thanks!

File: version.txt (hunk 3 above)
Comment (hakman) [reply version.txt]: done

File: hack/tag-release.sh (hunk 2 above)
Comment (hakman) [reply hack/tag-release.sh]: done

**Comment by hakman** [comment]
@vteratipally @mmiranda96 Any thoughts about this PR?

**Comment by hakman** [comment]
@vteratipally @Random-Liu Could you take one more look at this? Would help with releases going forward.

**Comment by disconn3ct** [comment]
Does the k8s org have a code of conduct for maintainers? What is the process for handling abdicated repositories?

**Pull request merged**